rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,361 @@
+```
+**Scheduled Trip (departure time from origin)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `departureTime` (Time): when the vehicle is scheduled to depart the given stop
+- `startStop` (string): GTFS stop ID the trip is departing from (either `location_type` 0 or 1)
+```json
+`{"serviceDate": "2023-01-20", "departureTime": "25:01:02", "startStop": "place-lake"}
+```
+**Scheduled Trip (arrival time at destination)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `arrivalTime` (Time): when the vehicle is scheduled to arrive at the given stop
+- `endStop` (string): GTFS stop ID the trip is ending at (either `location_type` 0 or 1)
+```json
+`{"serviceDate": "2023-01-20", "arrivalTime": "09:10:11", "endStop": "place-lake"}
+```
+**Scheduled Trip (known GTFS trip ID)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `gtfsId` (string): GTFS trip ID
+```json
+`{"serviceDate": "2023-01-20", "gtfsId": "123456790"}
+```
+
+### Trip Type
+When adding a trip, there are three kinds of trips which can be added:
+- `"arrival"`: a trip arriving to `endStop` at `time`
+- `"departure"`: a trip departing from `startStop` at `time`
+- `"roundTrip"`: a trip departing at `time` and expected to return to `startStop`
+
+## Events
+### Schema
+All events should have their schema documented in the [mbta/schemas](https://github.com/mbta/schemas) GitHub repository.
+
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They do this by either physically signing a paper trainsheet, or by typing their badge number into Glides. This event represent either type of signin, but may not be generated in the case where an operator only signs into the paper trainsheet.
+
+Event type: `com.mbta.ctd.glides.operator-signed-in.v1`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (Operator): the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+
+### Trip added
+This creates a new trip in the timesheet. It may or may not be a new trip relative to GTFS (see the splitting a train reference example).
+
+Event type: `com.mbta.ctd.glides.trip-added.v1`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): provides a unique identifier for the newly created trip
+- `startStop` (string, conditionally required): the GTFS stop ID at which the trip will be starting (either `location_type` 0 or 1). Required if `tripType` is `"departure"` or `"roundTrip"`.
+- `endStop` (string, conditionally required): the GTFS stop ID at which the trip will be ending (either `location_type` 0 or 1). Required if `tripType` is `"arrival"`. If `tripType` is `roundTrip`, this is the last stop before the train turns around to return to `startStop`.
```

Comment:
The paper trainsheets include the route number, and in some cases, this can vary for scheduled trips to/from the same terminal (eg, https://glides.mbta.com/trainsheet-export?service_date=2023-01-23&station=place-mdftf). Are there near-term plans for adding this to the added-trips UI, that might warrant adding it to the spec on the sooner side?

Reply:
I would be in favor of having it in the spec. We haven't included or planned endpoint changes in the UI because it's not in paper trainsheets so we didn't need to do it to launch. But it's basic information about a trip, and our backend is set up to do it whenever we get around to designing the input for it.

Reply:
I'll add a field for this.